Source/SwiftLintFramework/Rules/Style/UnneededParenthesesInClosureArgumentRule.swift

```diff
@@ -60,75 +75,74 @@ public struct UnneededParenthesesInClosureArgumentRule: ConfigurationProviderRul
         ]
     )
 
-    public func validate(file: SwiftLintFile) -> [StyleViolation] {
-        return violationRanges(file: file).map {
-            StyleViolation(ruleDescription: Self.description,
-                           severity: configuration.severity,
-                           location: Location(file: file, characterOffset: $0.location))
+    public func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor? {
+        Visitor()
```

Comment:
I like this pattern to have nested Visitor/Rewriter types to simplify the naming for these visitors and rewriters.